test/stdlib/Result.swift

```diff
@@ -0,0 +1,155 @@
+// RUN: %target-run-simple-swift
+// REQUIRES: executable_test
+
+import StdlibUnittest
+import Swift
+
+let ResultTests = TestSuite("Result")
+
+fileprivate enum Err: Error, Equatable {
+  case err
+  case derr
+}
+
+fileprivate let string = "string"
+
+ResultTests.test("Construction") {
+  let result1: Result<String, String> = .success(string)
+  let result2: Result<String, String> = .failure(string)
+  let string1: String? = {
+    switch result1 {
+      case let .success(string): 
+        return string
+      case .failure: 
+        return nil
+    }
+  }()
+  let string2: String? = {
+    switch result2 {
+      case let .failure(string): 
+        return string
+      case .success: 
+        return nil
+    }
+  }()
+
+  expectEqual(string1, string2)
+}
+
+ResultTests.test("Properties") {
+  let result1: Result<String, String> = .success(string)
+  let result2: Result<String, String> = .failure(string)
+
+  expectEqual(result1.value, result2.error)
```

Comment:
Likewise, the test could pass if both properties returned `nil`.

You should also test what `error` returns on a successful `Result` and what `value` returns on a failing `Result`.

Reply:
I've implemented more comparisons for these tests.